hack/e2e/environment.sh

```diff
@@ -0,0 +1,40 @@
+# File used as metal3-dev-env config and as a source for variables used by envsubst
+
+export IMAGE_OS="Ubuntu"
+export CONTAINER_RUNTIME="docker"
+export EPHEMERAL_CLUSTER="kind"
+export CAPI_VERSION="v1alpha3"
+export CAPM3_VERSION="v1alpha4"
+export NUM_NODES=4
+
+export KUBERNETES_VERSION=v1.21.1
+export API_ENDPOINT_HOST=192.168.111.249
+export API_ENDPOINT_PORT=6443
+export POD_CIDR="192.168.0.0/18"
+export SERVICE_CIDR="10.96.0.0/12"
+export CLUSTER_NAME=test1
+export NAMESPACE=metal3
+
+export PROVISIONING_POOL_RANGE_START=172.22.0.100
+export PROVISIONING_POOL_RANGE_END=172.22.0.200
+export PROVISIONING_CIDR=24
+
+export BAREMETALV4_POOL_RANGE_START=192.168.111.100
+export BAREMETALV4_POOL_RANGE_END=192.168.111.200
+export EXTERNAL_SUBNET_V4_PREFIX=24
+export EXTERNAL_SUBNET_V4_HOST=192.168.111.1
+export IMAGE_USERNAME=metal3
+export NODE_DRAIN_TIMEOUT=0s
+
+export PROVISIONING_URL_HOST="172.22.0.1"
+export IMAGE_NAME="UBUNTU_20.04_NODE_IMAGE_K8S_${KUBERNETES_VERSION}.qcow2"
+export IMAGE_LOCATION="https://artifactory.nordix.org/artifactory/airship/images/k8s_${KUBERNETES_VERSION}"
+export IMAGE_CHECKSUM="http://${PROVISIONING_URL_HOST}/images/UBUNTU_20.04_NODE_IMAGE_K8S_${KUBERNETES_VERSION}-raw.img.md5sum"
+export IMAGE_CHECKSUM_TYPE=md5
+export IMAGE_URL="http://${PROVISIONING_URL_HOST}/images/UBUNTU_20.04_NODE_IMAGE_K8S_${KUBERNETES_VERSION}-raw.img"
+
+# preserve the template, to be substituted based on value defined in e2e_test.go
+# shellcheck disable=SC2016
+export CONTROL_PLANE_MACHINE_COUNT='${CONTROL_PLANE_MACHINE_COUNT}'
```

Comment:
Hm, I can define the variable and have it substituted in envsubst. But I'm not sure why is decoupling from e2e_test.go better?

Reply:
Because, that way it is easier IMO to set number of controlplane and worker machines for the user rather than looking for go code.

Reply:
Ok, I've reverted that change. Although it means that the number given in e2e_test.go will be a bit confusing; it can be even set to a random number like `5` and the test will still pass alright.

Reply:
Judging by today's call adjusting it in the template is not easier than in go. Maybe let's restore the `CONTROL_PLANE_MACHINE_COUNT`?

Reply:
Yes, let's restore it.